core/rawdb/accessors_chain.go

```diff
@@ -976,3 +976,34 @@ func WritePendingEpoch(tx kv.RwTx, blockNum uint64, blockHash common.Hash, trans
 	copy(k[8:], blockHash[:])
 	return tx.Put(kv.PendingEpoch, k, transitionProof)
 }
+
+// Transitioned returns true if the block number comes after POS transition
+func Transitioned(db kv.Getter, blockNum uint64) (trans bool, err error) {
+	data, err := db.GetOne(kv.HeaderTD, dbutils.HeaderKey(math.MaxUint64, common.Hash{}))
+	if err != nil {
+		return false, fmt.Errorf("failed ReadTd: %w", err)
+	}
+	if len(data) == 0 {
+		return false, nil
+	}
+	return blockNum >= binary.BigEndian.Uint64(data), nil
+}
+
+// MarkTreansition sets transition to proof-of-stake from the block number
+func MarkTransition(db kv.StatelessRwTx, blockNum uint64) error {
+	data := make([]byte, 8)
+	binary.BigEndian.PutUint64(data, blockNum)
+	// If we already transitioned then we do not update the transition
+	marked, err := db.Has(kv.HeaderTD, dbutils.HeaderKey(math.MaxUint64, common.Hash{}))
```

Comment:
It is better not to overload the existing table (`kv.HeaderTD`) with the new meaning, but instead introduce a separate table (even if it is going to contain 1 record). Fewer unwanted side-effects in the code that already interprets `HeaderTD` table